test/bench/etc/run_granular_benchmarks.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+/**
+ * This script runs all benchmarks in test/bench/lib/granular and merges their output
+ */
+
+const cp = require('child_process');
+const fs = require('fs/promises');
+const path = require('path');
+const { Task } = require('dbx-js-tools/packages/bson-bench');
+
+const BENCHMARK_REGEX = /(.*)\.bench\.js$/;
+const BENCHMARK_PATH = path.resolve(`${__dirname}/../lib/granular`);
+const DOCUMENT_ROOT = path.resolve(`${__dirname}/../documents`);
+const LIBRARY_PATH = path.resolve(`${__dirname}/../../../.`);
+(async () => {
+  // HACK : run one dummy task with the local bson to ensure it's available for subsequent suites
+  await new Task({
+    documentPath: path.resolve(`${DOCUMENT_ROOT}/binary_small.json`),
+    library: `bson:${LIBRARY_PATH}`,
+    iterations: 1,
+    warmup: 1,
+    operation: 'deserialize',
+    options: {}
+  })
+    .run()
+    .catch(() => null);
+
+  // Run all benchmark files
+  const lib = await fs.readdir(BENCHMARK_PATH);
+  for await (const dirent of lib) {
+    if (BENCHMARK_REGEX.test(dirent)) {
+      const child = cp.fork(`${BENCHMARK_PATH}/${dirent}`);
+      if (child.stdout) child.stdout.pipe(process.stdout);
+      if (child.stderr) child.stdout.pipe(process.stderr);
+
+      await new Promise((resolve, reject) =>
+        child.once('exit', code => {
+          if (code !== 0) return reject();
+          return resolve();
+        })
+      );
+    }
+  }
+  const resultPaths = [];
+
+  for await (const dirent of await fs.opendir(__dirname)) {
+    if (/Results.json$/.test(dirent.name)) {
+      resultPaths.push(`./${dirent.name}`);
+    }
+  }
+
+  if (resultPaths.length === 0) throw new Error('Benchmarks did not run successfully');
+
+  // Ensure that there are no duplicate test-name/options pairs as this will prevent us from
+  // uploading with perf.send
+  const set = new Set();
+  for (const resultFile of resultPaths) {
+    const results = require(resultFile);
+    for (const entry of results) {
+      const name = entry.info.test_name;
```

Comment:
I'm not precisely following where it comes from but I think this is where the string like:

"timestamp_array_1000._deserialize_bson:/data/mci/20eab6d75d6f8a3717b1f015dd7a7648/src"

is coming from, don't we need this name to be consistent so that the trend chart can build on the history? I imagine that commit hash is going to change, right?

Reply:
That's a really good catch. I didn't notice that. The test name joins the test file name, the operation and the library used. We can set the test configurations to install from the main branch and that should have us covered as that won't change. That should give us a test name like `timestamp_array_1000._deserialize_bson#main`.